### DISCUSSION TITLE
aws - route53 - identify records pointing to dangling resources

### DISCUSSION THREAD

**Discussion by nitrocode** [cloud-custodian/cloud-custodian#7368] [General]
Unsure if this is a feature request or question but I'm trying to identify route53 records that currently point to stale resources such as deleted load balancers, removed and no longer used EIPs, etc.

Can this be done already ? 

If not, how can we extend functionality to provide this ? Perhaps an "unused" filter that will check for stale resources and return hosted zones and records that could be removed ?

**Comment by kapilt** [comment]
for non alias resources r53 could be pointing to an arbitrary ip or dns name with no correlation to other resources in the account.

for alias resources (load balancer, api gateway, s3, cloud front, etc), we could sanity check the alias target exists. 
[0] - https://docs.aws.amazon.com/Route53/latest/APIReference/API_AliasTarget.html

EIPs aren't aliases though, they would be an A record which isn't something we could distinguish if the EIP is in a different account, released, etc. For a private dns on A records things also get interesting wrt to peers / shared vpcs / route tables, ie defining an A record for datacenter resource.

**Comment by nitrocode** [comment]
You mentioned:
> for alias resources (load balancer, api gateway, s3, cloud front, etc), we could sanity check the alias target exists.
[0] - https://docs.aws.amazon.com/Route53/latest/APIReference/API_AliasTarget.html

This makes sense to me.

> for non alias resources r53 could be pointing to an arbitrary ip or dns name with no correlation to other resources in the account.

That makes sense too but we could create an `invalid` filter like `aws.asg.filters.invalid` using similar logic in [Cimpress-MCP/red-x](https://github.com/Cimpress-MCP/red-x ) to find...

> abandoned/misconfigured domain delegations or records pointing at inactive domains from managed services
